packages/extension/src/CodeDialog/CodeDialog.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormControl,
+  FormHelperText,
+  TextField
+} from "@material-ui/core";
+import React from "react";
+
+interface CodeDialogProps {
+  value?: string;
+  open: boolean;
+  onClose: () => void;
+  onSubmit: (value: string) => void;
+}
+
+const CodeDialog: React.SFC<CodeDialogProps> = (props: CodeDialogProps) => {
+  const { open, onClose, onSubmit } = props;
+
+  const [value, setValue] = React.useState<string>("");
+
+  const [isValid, setIsValid] = React.useState(false);
+
+  const [lastValue, setLastValue] = React.useState<string>();
+
+  if (props.value != null && lastValue !== props.value) {
+    setValue(props.value);
+    setLastValue(props.value);
+  }
+
+  const handleInputChanged = React.useCallback(
+    val => {
+      setValue(val);
+      setIsValid(true);
+    },
+    [value, setValue, setIsValid]
+  );
+
+  const reset = () => {
+    setValue("");
+  };
+
+  const handleCancel = React.useCallback(() => {
+    reset();
+    onClose();
+  }, [setValue, onClose]);
+
+  const handleSubmit = React.useCallback(() => {
+    reset();
+    onSubmit(value);
+  }, [value, setValue, onSubmit]);
+
+  return (
+    <Dialog
+      maxWidth="md"
+      fullWidth={true}
+      open={open}
+      onClose={onClose}
+      aria-labelledby="form-dialog-title"
+    >
+      <DialogTitle id="form-dialog-title">Code block</DialogTitle>
+      <DialogContent>
+        <FormControl fullWidth={true}>
+          <TextField
+            autoFocus={true}
+            id="href"
+            label="Language"
+            type="input"
+            fullWidth={true}
+            value={value}
```

Comment:
There is no validation for this value. If it contains a `, the markdown output will be invalid.

You should look at the anchor dialog in the tools branch to see how this can be done.

Reply:
Done